src/classes/CRLP_Rollup_BATCH_PSC_LDV.cls

```diff
@@ -0,0 +1,370 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2017
+* @group Rollups
+* @group-content ../../ApexDocContent/Rollups2.htm
+* @description Soft Credits rollup to the Account and Contacts object using OpportunityContactRole as the base
+*  LDV style query. Stateful vars are used to hold rollups of the Account and Contact across multiple batch
+*  iterations.
+*/
+public class CRLP_Rollup_BATCH_PSC_LDV implements Database.Batchable<SObject>, Database.Stateful {
+
+    /*******************************************************************************************************
+    * @description Stateful collection of Rollups used to hold Rollups for Contacts and Accounts that
+    * span beyond a single batch iteration
+    */
+    private List<CRLP_Rollup> cntctStatefulRollups;
+    private List<CRLP_Rollup> acctStatefulRollups;
+    private Set<Id> acctStatefulOppIds;
+    private Id currentAccountRecordId;
+    private Id currentContactRecordId;
+    private Boolean hasAccountSoftCreditRollups = false;
+    private Boolean hasContactSoftCreditRollups = false;
+    private List<Id> recordIds;     // Specified list of Summary (Account or Contact) Id's to roll up
+
+    /*******************************************************************************************************
+    * @description Constructs the class, sets the properties
+    */
+    public CRLP_Rollup_BATCH_PSC_LDV() {
+        this.cntctStatefulRollups = new List<CRLP_Rollup>();
+        this.acctStatefulRollups = new List<CRLP_Rollup>();
+    }
+
+    /*******************************************************************************************************
+    * @description Constructs the class, sets the properties
+    */
+    public CRLP_Rollup_BATCH_PSC_LDV(List<Id> recordIds) {
+        this.recordIds = recordIds;
+        this.cntctStatefulRollups = new List<CRLP_Rollup>();
```

Comment:
Super nit-pick, but I like just calling the previous constructor so that it's abundantly clear what we're doing differently. Also makes additions to the other constructor a bit easier.